File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    static ArrayList<Integer> levels = new ArrayList<Integer> ();

    
    public void backTrack(TreeNode node, int depth){

        if (node == null) return;
        
        if (depth == levels.size()){
            levels.add(node.val);
        } else {
           levels.set(depth, node.val);
        }

        backTrack(node.left, depth+1);
        backTrack(node.right, depth+1);
    }

    public List<Integer> rightSideView(TreeNode root) {
        levels.clear();
        backTrack(root, 0);
        List<Integer> result = levels;
        return result;
    }
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        
        levels = []
        
        def backTrack(node, depth):

            if not node:
                return
            
            if len(levels) == depth:
                levels.append(node.val)
            else:
                levels[depth] = node.val
            
            backTrack(node.left, depth+1)
            backTrack(node.right, depth+1)
        
        backTrack(root, 0)
        
        return levels